src/rgw/driver/rados/rgw_bucket.cc

```diff
@@ -2648,6 +2725,21 @@ int RGWMetadataHandlerPut_BucketInstance::put_post(const DoutPrefixProvider *dpp
     }
   } /* update lc */
 
+  /* update bucket topic mapping */
+  {
+    auto* orig_obj = dynamic_cast<RGWBucketInstanceMetadataObject*>(old_obj);
```

Comment:
why "dynamic_cast"?
in other places we use "static_cast". if casting is not guranteed we probably have another isssue.